vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,403 @@
+#> <environment: 0x559280451700>
+#> 
+
+```
+
+In the above example, if you have some reason to believe that the standard deviation of the means used could we smaller, you could reduce them by half to 0.5.
+
+```r
+
+new_priors <- default_priors$priors
+
+new_priors[ ,sd := 0.5]
+
+```
+
+You could then pass these new priors to the `epinowcast` function.
+
+```r
+epinowcast(pobs,
+  expectation = expectation_module,
+  fit = fit,
+  model = multithread_model,
+  priors = new_priors
+)
+```
+Similarly, you could take an emperical Bayes approach and use the posteriors of a [fitted model as described in another vignette](https://package.epinowcast.org/articles/germany-age-stratified-nowcasting.html#using-the-inflated-posterior-as-a-prior).
+
+
+## Exploring your data {.modelData}
+
+<!--
+#TODO Non-stationarity, longer time scales, multi-modality from known etiologies
+-->
+
+It is important to understand the data being passed to `epinowcast` and have some ideas regarding the data generating process for your data.
+For instance, if there are significant changes in your reporting triangle, this could manifest as multi-modality in the posterior distribution (i.e., Stan is trying to fit two different reporting delays with the same parameters).
+These different reporting regimes could result in long model run times, very wide posteriors, and ultimately nonsensical inferences.
+Given the nuances of your data, you may need to identify and fit different regimes of your data.
+These problems can also manifest over longer timescales, so while more data are generally better, too much data can result in ill-fitting models.
+
+
+## Common failure modes
+
+### My model takes too long to run
+
+One of the more common issues with Bayesian inference is that the sampler may take a long time to run.
+This can occur because the sampler is having difficulty exploring the posterior parameter space given the model and your data.
+You can a few simple steps to assist with this:
+
+
+1. Increase the [`adapt_delta`](.adaptDelta) argument to 15 which will allow longer trajectories to used for steps when sampling. This is especially important if the `per_at_max_treedepth` value is high in the returned CmdStan diagnostics.
+2. Use more [informative priors](.settingPriors). 
+3. Try to simplify the model first to understand potential degeneracies.
+4. Reduce the amount of data you are trying to fit. This might include reducing the duration of time overwhich you are trying to fit. Once you can confirm that the model will fit your data, it may just be a question of computational power when fitting the larger data. 
+
+
+### Divergent transitions 
+
+CmdStan may alert you that you have ["divergent transitions"](https://mc-stan.org/docs/reference-manual/divergent-transitions.html) detected during your model fit.
+
+```r
+#>1: There were 32 divergent transitions after warmup.
+```
+
+A high number of divergent transitions indicates that the model outputs are not to be trusted.
```

Comment:
I think it's probably worth giving a number here for "high" (or a proportion of samples)

Reply:
This is also a bit subjective so whilst I think we can bit a bit clearer I don't think we should give a very specific value.